sqlalchemy_jsonapi/declarative/serializer.py

```diff
@@ -0,0 +1,118 @@
+"""A serializer for serializing SQLAlchemy models to JSON API spec."""
+
+import datetime
+
+from inflection import dasherize, underscore
+
+
+class JSONAPISerializer(object):
+    """A JSON API serializer that serializes SQLAlchemy models."""
+
+    def serialize(self, resources):
+        """Serialize resource(s) according to json-api spec."""
+        try:
+            self.fields
+            self.model
+            self.dasherize
+        except AttributeError:
+            raise
```

Comment:
If the serializer *must* have these in order to operate, `JSONSerializer` should come with sane defaults, so that we don't have to check if they've been implemented by the user.

Imagine finding a new library, and implementing a class from that library, and then your code raising an `AttributeError` for code that lives inside the thing you're subclassing. You're going to think "This library is broken..." (or, at least, that's what I'd think); that'd be partially correct, in that the API has this sharp edge that you've now cut yourself on, but you also weren't implementing it the way that it "should" to be implemented.